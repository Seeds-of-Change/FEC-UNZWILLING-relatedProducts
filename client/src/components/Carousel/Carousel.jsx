import React from 'react';
import PropTypes from 'prop-types';
import style from './styles.css';
import ProductCard from '../ProductCard/ProductCard';

export default class Carousel extends React.Component {
  constructor(props) {
    super(props);

    this.scrollRef = React.createRef();
    this.goToPrevious = this.goToPrevious.bind(this);
    this.goToNext = this.goToNext.bind(this);
  }

  goToPrevious() {
    this.scrollRef.current.scrollLeft -= 175;
  }

  goToNext() {
    this.scrollRef.current.scrollLeft += 175;
  }

  render() {
    const { products, updateProduct } = this.props;
    return (
      <div className={style.wrapper}>
        <button type="button" className={`${style.arrows} ${style.arrows_left}`} onClick={this.goToPrevious}>
          <span />
        </button>

        <div className={style.cards} ref={this.scrollRef}>
          {products.map((product) => (
            <ProductCard
              product={product}
              updateProduct={updateProduct}
              className={style.slide_item}
            />
          ))}
        </div>

        <button type="button" className={`${style.arrows} ${style.arrows_right}`} onClick={this.goToNext}>
          <span />
        </button>

      </div>
    );
  }
}

Carousel.defaultProps = {
  products: [],
  updateProduct: null,
};

Carousel.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      brand: PropTypes.string,
      name: PropTypes.string,
      product_features:
        PropTypes.shape({
          header: PropTypes.string,
          features: PropTypes.arrayOf(
            PropTypes.shape({
              heading: PropTypes.string,
              description: PropTypes.string,
              posX: PropTypes.number,
              posY: PropTypes.number,
            }),
          ),
        }),
      related_products: PropTypes.arrayOf(PropTypes.number),
      image: PropTypes.string,
      imageMini: PropTypes.string,
      price: PropTypes.number,
      salePrice: PropTypes.number,
      onSale: PropTypes.bool,
      rating: PropTypes.number,
      isFavorite: PropTypes.bool,
    }),
  ),
  updateProduct: PropTypes.func,
};
