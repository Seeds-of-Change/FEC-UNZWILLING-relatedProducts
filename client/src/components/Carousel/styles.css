.wrapper {
  background: #F2F0EE;
  position: relative;
}

.cards {
  margin: 100px auto;
  max-width: 80%;
  overflow-x: hidden;
  overflow-y: hidden;
  white-space: nowrap;
}

.cards.active {
  display: block;
}

button.arrows {
  position: absolute;
  border: none;
  padding: 0;
  outline:inherit;
  width: 50px;
  height: 50px;
  line-height: 50px;
  top: 40%;
  cursor: pointer;
}

.arrows:focus {
  outline: 0;
}

.arrows:hover {
  background: white;
  border-color: white;
  color: #111;
  opacity: .5;
}

.arrows_left {
  left: 5%;
}

.arrows span{
  display:block;
  height:25px;
  width:25px;
}
.arrows_left > span {
  border-bottom: 2px solid;
  border-left: 2px solid;
  transform: rotateZ(45deg);
}

.arrows_right {
  right: 5%;
}
.arrows_right > span {
  border-top: 2px solid;
  border-right: 2px solid;
  transform: rotateZ(45deg);
}

.arrows:after {
  position: absolute;
  display: inline-block;
  content: "";
  width: 25px;
  height: 25px;
  top: 50%;
  left: 50%;
}
