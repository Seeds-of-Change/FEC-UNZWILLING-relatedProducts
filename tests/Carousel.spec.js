import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from '../client/src/components/Carousel/Carousel';

describe('Carousel', () => {
  describe('Have props', () => {
    let component;

    beforeEach(() => {
      const { products } = [
        {
          id: 1,
          brand: 'Zwilling ENFINIGY',
          name: 'Power Blender',
          product_features: {},
          related_products: [1, 3, 15, 21],
          image: 'https://unzwilling.s3-us-west-1.amazonaws.com/1.jpg',
          imageMini: 'https://unzwilling.s3-us-west-1.amazonaws.com/1_mini.jpg',
          price: 100,
          salePrice: 80,
          onSale: true,
          rating: 4.5,
          isFavorite: true,
        },
        {
          id: 2,
          brand: 'Zwilling ENFINIGY',
          name: 'Power Blender',
          product_features: {},
          related_products: [1, 3, 15, 21],
          image: 'https://unzwilling.s3-us-west-1.amazonaws.com/1.jpg',
          imageMini: 'https://unzwilling.s3-us-west-1.amazonaws.com/1_mini.jpg',
          price: 100,
          salePrice: 80,
          onSale: true,
          rating: 4.5,
          isFavorite: true,
        },
        {
          id: 3,
          brand: 'Zwilling ENFINIGY',
          name: 'Power Blender',
          product_features: {},
          related_products: [1, 3, 15, 21],
          image: 'https://unzwilling.s3-us-west-1.amazonaws.com/1.jpg',
          imageMini: 'https://unzwilling.s3-us-west-1.amazonaws.com/1_mini.jpg',
          price: 100,
          salePrice: 80,
          onSale: true,
          rating: 4.5,
          isFavorite: true,
        },
      ];

      component = renderer.create(
        <Carousel products={products} />,
      );
    });

    test('snapshot renders', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
