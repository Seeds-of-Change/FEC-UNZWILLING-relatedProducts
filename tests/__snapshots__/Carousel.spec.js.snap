// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Have props snapshot renders 1`] = `
<div
  className="wrapper"
>
  <button
    className="arrows arrows_left"
    onClick={[Function]}
    type="button"
  >
    <span />
  </button>
  <div
    className="cards"
  />
  <button
    className="arrows arrows_right"
    onClick={[Function]}
    type="button"
  >
    <span />
  </button>
</div>
`;
